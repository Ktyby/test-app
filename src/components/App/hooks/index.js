import useModal from "./useModal";
import useHandleCreateUser from "./useHandleCreateUser";

export { useModal, useHandleCreateUser };
