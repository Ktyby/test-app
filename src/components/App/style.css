.main {
  display: flex;
  flex-direction: column;
  align-items: center;

  margin-top: 80px;
}

.main__button {
  width: 150px;
  height: 35px;

  margin-bottom: 40px;

  border: none;
  border-radius: 10px;

  cursor: pointer;
  transition: 0.4s;
}

.main__button:hover {
  background-color: #000;
  color: #fff;
}

.main__paginate-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.main__paginate {
  display: flex;
  justify-content: space-between;
  padding: 0;
  width: 250px;

  color: #fff;
}

.main__paginate > li {
  transition: 0.4s;
  cursor: pointer;
}

.main__paginate > li:hover {
  color: #fcd911;
}
