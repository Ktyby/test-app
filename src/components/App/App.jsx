import React from "react";

import Form from "../Form";

import "./style.css";

const App = () => {
  return (
    <div className="main">
      <button className="main__button">Create person</button>
      <Form />
    </div>
  );
};

export default App;
