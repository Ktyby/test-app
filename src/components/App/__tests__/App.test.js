import renderer from "react-test-renderer";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import App from "../App";

const mockStore = configureMockStore();

describe("components => App", () => {
  const store = mockStore({
    users: [],
  });

  it("should render App component", () => {
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
