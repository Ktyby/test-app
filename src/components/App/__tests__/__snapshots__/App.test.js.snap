// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components => App should render App component 1`] = `
<div
  className="main"
>
  <button
    className="main__button"
    onClick={[Function]}
  >
    Create person
  </button>
  <div
    className="main__paginate-container"
  >
    <table>
      <tbody />
    </table>
  </div>
</div>
`;
